addon/services/tour.js

```diff
@@ -222,55 +230,59 @@ export default Service.extend(Evented, {
    * @public
    */
   addSteps(steps) {
-    this._initialize();
-    const tour = get(this, 'tourObject');
-
-    // Return nothing if there are no steps
-    if (isEmpty(steps)) {
-      return;
-    }
-    /* istanbul ignore next: also can't test this due to things attached to root blowing up tests */
-    if (!this._requiredElementsPresent()) {
-      tour.addStep('error', {
-        buttons: [{
-          text: 'Exit',
-          action: tour.cancel
-        }],
-        title: get(this, 'errorTitle'),
-        text: [get(this, 'messageForUser')]
-      });
-      return;
+    if (this.fastboot) {
```

Comment:
I would wrap the entire method in `if(!this._isFastBoot)`, rather than resolving a promise here.

Reply:
I think we do need a promise here - or we should change the way shepherd is loaded. For instance:
```
this.tour.addSteps([
      {
      }
]).then(() => {
  this.tour.start();
})
```
will blow up in fastboot if we do not return a valid promise here. We could reject the promise, but that might require the user to properly handle errors in the app when in fastboot mode.
We could of course add a `load()` method and make that behaviour more explicit (and remove the promise from `addSteps()`)